lisp/ess-inferior-r.el

```diff
@@ -0,0 +1,250 @@
+;;; ess-inferior-r.el --- inferior-r-process         -*- lexical-binding: t; -*-
```

Comment:
I like the new file but it is very important to me that we keep the full git history by splitting files the hard way as outlined in https://github.com/emacs-ess/ESS/pull/606#issuecomment-431657961.

I have been guilty of butchering the history in past refactorings, which I regret.

Reply:
The more I work with it the less I'm sure about splitting this into a new file. The inferior stuff is pretty tightly integrated with the "normal" major mode stuff. It forces us to put stuff like:

```elisp
;; ensure `R' is defined if the user didn't load the autoloads file
(autoload 'R "ess-inferior-r")
(autoload 'run-ess-r- "ess-inferior-r")
```

in ess-r-mode.el, which is a little odd. What do you think about keeping this all in ess-r-mode.el?

Reply:
Sounds good, we can always split later on if needed.